Satoshi/sightings form

sgmbr opened the pull request:
Sightings / Non-sightings forms implemented

sgmbr commented on src/components/report/ReportNonSighting.js:
```diff
@@ -0,0 +1,93 @@
+    communications: false,
+  },
+  comments: '',
+};
```
Initial value for every field should be passed to Formik or otherwise React complains when I input some value that I'm changing uncontrolled input to controlled.

sgmbr commented on src/components/helpers/RenderField.js:
```diff
@@ -0,0 +1,54 @@
+import React from 'react';
```
Your renderField helper was very good. Just tweaked a little bit.

sgmbr commented on src/components/report/ReportType.js:
```diff
@@ -4,32 +4,32 @@ import { Link } from 'react-router-dom';
 const ReportType = props => {
   return (
     <div className="ReportType">
```
Just fixed indentation and bootstrap class names

sgmbr commented on src/actions/helpers/formatSighting.js:
```diff
@@ -0,0 +1,66 @@
+import moment from 'moment';
```
Not formatting passed form values because that would change the values in the form. Creating a new object instead for POST.

georgemoon commented on src/components/report/ReportNonSighting.js:
```diff
@@ -0,0 +1,93 @@
+    name: yup.string().required(requiredMessage),
+    email: yup.string().email('Invalid email address').required(requiredMessage),
+  }),
+});
```
Are validation results from the backend still showing up, or is it front-end only?

sgmbr replied:
Fixed to show backend validation results when available

sgmbr commented:
Fixed to show back end validation results when available.

Also added more granular front end validation for point_location (type, min, max) (I thought I've already done this but not)
